newIDE/app/src/ProjectCreation/CreateProjectDialog.js

```diff
@@ -8,9 +8,15 @@ import { Tabs, Tab } from '../UI/Tabs';
 import Tutorials from './Tutorials';
 import { Column } from '../UI/Grid';
 import { type StorageProvider, type FileMetadata } from '../ProjectsStorage';
+import { findEmptyPath } from './LocalPathFinder';
```

Comment:
So this works :) The downside is that the web-app is now depending in LocalPathFinder, which is a bit sad because any LocalXXX file should not be used by the web-app (web-app = BrowserApp.js).

Reply:
Is that such a big deal? That file is very small, and I don't see any "react way" of solving this problem, because that would mean managing the state of the component from the children as those are electron only, but, quote from you,
> You're not sharing actually, you're giving "just enough stuff" for your children component to do its job.

So sharing the state for children for letting them handle/set it would be an anti-pattern?

Reply:
You're not forced to duplicate the state. Instead you can use the second solution I explained here: https://github.com/4ian/GDevelop/pull/1763#discussion_r427806624

You can the state in the parent but set it empty, and use a componentDidMount in the children to change the path (see my answer).

The third solution is to make a function that we pass in LocalApp/BrowserApp that is a function that returns the default path - but at this point we're complicating stuff too much for something so boring. 
Just put two "componentDidMount", that are setting the output path if it's empty using the prop to change it and you're done.